### DISCUSSION TITLE
my ips provider doesn't allow port forwarding

### DISCUSSION THREAD

**Discussion by biscqit** [nextcloud/all-in-one#5143] [Questions] [answered]
what system should I use.

also please spam this with the most amount of resources you have on everything that has to do with this.

thanks.

**Comment by szaimen** [comment, marked as answer]
Hi, I see three options for you:
1. https://github.com/nextcloud/all-in-one#how-to-run-nextcloud-behind-a-cloudflare-tunnel
2. https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-get-nextcloud-running-using-the-acme-dns-challenge
3. Buy a certificate and use that:
If none of the above ways work for you, you may simply buy a certificate from an issuer for your domain. You then download the certificate onto your server, configure AIO in [reverse proxy mode](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md) and use the certificate for your domain in your reverse proxy config.

**Reply by biscqit** [reply]
so thing is I can port forward but its really weird:
![image](https://github.com/user-attachments/assets/8c0b2b96-59d6-4df9-b4a0-66f1ff5c54f3)

that's the only way I can do it, if i put 80/443 in both internal and external it won't activate the rule

**Reply by biscqit** [reply]
sorry for unmarking just curious as to what you might think
